registration/api/v2/views.py

```diff
@@ -401,6 +401,22 @@ def static_content_overrides(self, request, pk=None):
 
             return Response(status=status.HTTP_200_OK, data=application.draft_static_content_overrides)
 
+    def list(self, request, *args, **kwargs):
+        """
+        Endpoint for getting the list of instances owned by the current user
+
+        If `fail_if_staff` is set for a staff user, the request returns a 400 error. This is used by the frontend to
+        prevent staff users from accidentally accessing and updating other users' instances.
+        """
+        if request.query_params.get("fail_if_staff", False) and request.user.is_staff:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'details': "Not allowed for staff users"
```

Comment:
Can we also add a code; something like `code`: 'NOT_ACCESSIBLE_TO_STAFF'` so that on the frontend we can match on the code instead of the error message string? It's not that important, but it would feel a bit cleaner to not have to rely on the exact error message.